tests/drivers/test_cryomagnetics_4g.py

```diff
@@ -0,0 +1,101 @@
+import io
+import logging
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from qcodes.instrument_drivers.cryomagnetics.Cryomagnetics4G_visa import (
+    Cryomagnetics4GException,
+    CryomagneticsModel4G,
+)
+
+# here the original test has a homemade log system that we don't want to
+# reproduce / write tests for. Instead, we use normal logging from our
+# instrument.visa module
+iostream = io.StringIO()
+logger = logging.getLogger("qcodes.instrument.visa")
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(created)s - %(message)s")
+lh = logging.StreamHandler(iostream)
+logger.addHandler(lh)
+lh.setLevel(logging.DEBUG)
+lh.setFormatter(formatter)
```

Comment:
This looks like copy paste from the AMI driver? I don't think we want to continue modifying the logging manually like this. Rather we should use pytests build in caplog figure to capture logs as needed